eds_scikit/io/hive.py

```diff
@@ -282,21 +288,22 @@ def persist_tables_to_folder(
         # to disk. Set a limit on the number of patients in the cohort ?
 
         if person_ids is not None:
-            person_ids = self._prepare_person_ids(person_ids)
+            person_ids = self._prepare_person_ids(person_ids, return_df=False)
+
+        connector = arrowConnector(
+            sql=self.spark_session.sql,
+            database_name=self.database_name,
+        )
 
         for table in tables:
             filepath = os.path.join(folder, f"{table}.parquet")
-            df = self._read_table(table, person_ids=person_ids)
-            self._write_df_to_parquet(df, filepath)
-
-    def _write_df_to_parquet(
-        self,
-        df: DataFrame,
-        filepath: str,
-    ) -> None:
-        assert os.path.isabs(filepath)
-        print(f"writing {filepath}")
-        df.to_pandas().to_parquet(filepath)
+            logger.info(f"Writing table {table} at {filepath}")
+            df = connector.get_pd_table(table_name=table, person_ids=person_ids)
+            df.to_parquet(
```

Comment:
See the comment about using `pq.dataset_writer` instead of loading a full table in RAM